samza-core/src/main/scala/org/apache/samza/container/TaskInstance.scala

```diff
@@ -210,7 +198,9 @@ class TaskInstance(
 
     trace("Flushing producers for taskName: %s" format taskName)
 
-    collector.flush
+    exceptionHandler.maybeHandle { // Allow users to decide whether to swallow producer exceptions (in kafka whole batch will be lost!)
+      collector.flush
```

Comment:
One question about this: now we wrap the flush with the exception handler, so we expect the user to provide their own to handle this exception? I am wondering what is the steps user needs to do if they want to do something here.